sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedPolicy.java

```diff
@@ -82,25 +82,71 @@ public static ChangeFeedPolicy createFullFidelityPolicy(Duration retentionDurati
         }
 
         ChangeFeedPolicy policy = new ChangeFeedPolicy();
-        policy.setFullFidelityRetentionDurationInMinutes((int)(retentionDuration.getSeconds() / 60));
+        policy.setAllVersionsAndDeletesRetentionDurationInMinutes((int)(retentionDuration.getSeconds() / 60));
         return policy;
     }
 
     /**
-     * Creates a default ChangeFeedPolicy without retention duration specified. With the default/incremental
+     * Creates a ChangeFeedPolicy with retention duration for {@link ChangeFeedMode#ALL_VERSIONS_AND_DELETES} processing
+     *
+     * @param retentionDuration  - the retention duration (max granularity in minutes) in which it
+     *                             will be possible to process change feed events with {@link ChangeFeedMode#ALL_VERSIONS_AND_DELETES} mode.
+     *
+     * @return ChangeFeedPolicy for {@link ChangeFeedMode#ALL_VERSIONS_AND_DELETES} change feed.
+     */
+    @Beta(value = Beta.SinceVersion.V4_37_0,
+        warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public static ChangeFeedPolicy createAllVersionsAndDeletesPolicy(Duration retentionDuration) {
+
+        if (retentionDuration.isNegative() ||
+            retentionDuration.isZero() ||
+            retentionDuration.getNano() != 0 ||
+            retentionDuration.getSeconds() % 60 != 0) {
+            throw new IllegalArgumentException(
+                "Argument retentionDuration must be a duration of a positive number of minutes."
+            );
+        }
+
+        ChangeFeedPolicy policy = new ChangeFeedPolicy();
+        policy.setAllVersionsAndDeletesRetentionDurationInMinutes((int)(retentionDuration.getSeconds() / 60));
```

Comment:
why not use duration.toMinutes()?

Reply:
Copied from existing API, agreed, we should use minutes directly.